src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,72 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class FigureSupplier {
+    private static final int MAX_RANDOM_SIZE = 100;
+    private static final double DEFAULT_SIZE = 10;
+    private static final int FIGURE_COUNT = 5;
+    private Random random = new Random();
+    private ColorSupplier supplier = new ColorSupplier();
+
+    public Figure getRandomFigure() {
+        int randomFigure = random.nextInt(FIGURE_COUNT);
+        switch (randomFigure) {
+            case 1:
+                return getRandomCircle();
+            case 2:
+                return getRandomRightTriangle();
+            case 3:
+                return getRandomIsoscelesTrapezoid();
+            case 4:
+                return getRandomRectangle();
+            default:
+        }
+        return getRandomSquare();
+    }
+
+    public Figure getRandomCircle() {
+        Circle circle = new Circle(random.nextInt(MAX_RANDOM_SIZE),
+                supplier.getRandomColor());
+        circle.getArea();
+        return circle;

```

Comment:
You could avoid creating local variable and strictly return new Circle
```suggestion
        return new Circle(random.nextInt(MAX_RANDOM_SIZE),
                supplier.getRandomColor());

```